src/Domain/FunctionsOfTime/QuaternionFunctionOfTime.cpp

```diff
@@ -68,7 +108,9 @@ void QuaternionFunctionOfTime<MaxDeriv>::update_stored_info() {
           << angle_deriv_info.size() << " stored angles and "
           << stored_quaternions_and_times_.size() << " stored quaternions.");
 
-  const size_t last_index = stored_quaternions_and_times_.size();
+  // We copy into a size_t so this whole function uses the same index
+  const size_t last_index =
+      stored_quaternion_size_.load(std::memory_order_relaxed);
```

Comment:
Not necessarily a change request, but why didn't you use a `size_t` to begin with?

Reply:
What do you mean by this? The datatype of `size_t` didn't change

Reply:
You're specifically commenting on the result type, so I assume it is significant somehow?

I assumed this was some commentary on the choice of `uint64_t` for the atomic type instead of `size_t` (despite them being the same everywhere we support), which did seem odd but not something that needed changing.

Reply:
Oh yeah I added `size_t` because of the `uint64_t`. And I used `uint64_t` because that's what @nilsdeppe suggested. I'm guessing it's just to be explicit so we avoid different behavior on different machines because we do padding in the hpp?

Reply:
Yea, in practice size_t and uint64_t are the same, but I've been a bit paranoid when it comes to atomics just to get some more control over what things translate into at the hardware level

Reply:
I just changed the wording of this comment to not use `size_t`